### DISCUSSION TITLE
Reconnect after Not connected.

### DISCUSSION THREAD

**Discussion by RogerG77** [pymodbus-dev/pymodbus#1646] [Q&A] [answered]
Hello,
I create a client in a blocking manner as follows and use it in further tasks. How can I implement a reconnect functionality if the connection to the PLC is lost ? (hadle after: Received ModbusException(Modbus Error: [Connection] Not connected[AsyncModbusTcpClient)

async def create_modbus_client() -> AsyncModbusTcpClient:
    modbus_client = AsyncModbusTcpClient(
        "192.168.0.1",
        port=502,
        framer=ModbusSocketFramer,
        timeout=10,
        retries=3,
        # retry_on_empty=True,
        # close_comm_on_error=False,
        # strict=True,
        # source_address=("localhost", 0),
    )
    await modbus_client.connect()
    assert modbus_client.connected
    return modbus_client

loop = get_event_loop()
modbus_client = loop.run_until_complete(loop.create_task(create_modbus_client()))
loop.create_task(modbus_writer(modbus_c, modbus_queue))
loop.create_task(modbus_reader(modbus_c, mqtt_queue))
loop.run_forever()

**Comment by janiversen** [comment, marked as answer]
you do not say which release you use?

is it the assert you have that fails or later ? pymodbus does "reconnect" meaning only after the first successful connect.

If you want us to look at what happens then please use "pymodbus_apply_log... as pr issue documentation so we can see what happens.

If you want to reconnect because the first connect fails, you simply need to call .connect(), if it happens later you need to do nothing.

**Reply by RogerG77** [reply]
Hello, I am using the 3.3.2 version. You are right, the reconnect is automatically done after connection was established with first connect.

**Comment by janiversen** [comment]
Looking at your program is seems very strange, why have a task for the first connect ?

I think it would help you looking at our examples like e.g. client_async.py
